crates/project/src/project.rs

```diff
@@ -3201,6 +3201,51 @@ impl Project {
         let disk_based_diagnostics_progress_token =
             adapter.disk_based_diagnostics_progress_token.clone();
 
+        // Log LSP messages under experimental/serverStatus and display pop-up for errors
+        language_server
+            .on_notification::<ServerStatus, _>({
+                let this = this.clone();
+                let name = name.to_string();
+                move |params, mut cx| {
+                    let this = this.clone();
+                    let name = name.to_string();
+                    match params.health {
+                        ServerHealthStatus::Ok => log::info!("Health: Ok"),
+                        ServerHealthStatus::Warning => log::warn!(
+                            "Health: Warning.\nMessage: {}",
+                            params.message.unwrap_or_default()
+                        ),
+                        ServerHealthStatus::Error => {
+                            log::error!(
+                                "Health: Error.\nMessage: {}",
+                                params.message.clone().unwrap_or_default()
+                            );
+                            let (tx, _rx) = smol::channel::bounded(1);
+                            let request = LanguageServerPromptRequest {
+                                level: match params.health {
```

Comment:
This match does not make any sense due to us entering into `ServerHealthStatus::Error` above.
Let's also ensure we do not send empty prompts (for `None` or empty trimmed `message).